vector/src/main/java/im/vector/app/features/home/room/detail/composer/voice/VoiceMessageViews.kt

```diff
@@ -348,8 +348,10 @@ class VoiceMessageViews(
     fun renderRecordingWaveform(amplitudeList: List<Int>) {
         views.voicePlaybackWaveform.doOnLayout { waveFormView ->
             val waveformColor = ThemeUtils.getColor(waveFormView.context, R.attr.vctr_content_quaternary)
+            val view = waveFormView as AudioWaveformView
+            view.clear()
```

Comment:
I may have found something here. The OOM occurs because `rawFftList` in `AudioWaveformView` is growing bigger and bigger (500,000 objects after 1 minute of recording already). I noticed that here we're redrawing the waveform without first clearing the view and that keeps all the previous values in `rawFftList` alive. With the clear in place, I'm not getting any OOMs anymore neither in the emulator nor on my device.

@mnaturel could you verify this?

Reply:
Good catch! It was effectively what was causing the memory issue. I could record 15 minutes without problem. I am wondering if we should also call the clear() method in `MessageVoiceItem`. I would update the unbind method as follow:
```
override fun unbind(holder: Holder) {
    holder.voicePlaybackWaveform.clear()
    contentUploadStateTrackerBinder.unbind(attributes.informationData.eventId)
    contentDownloadStateTrackerBinder.unbind(mxcUrl)
    audioMessagePlaybackTracker.untrack(attributes.informationData.eventId)
    super.unbind(holder)
}
```

Reply:
Oh, probably a good idea, yeah. Made the change.